src/components/Chip/Chip.tsx

```diff
@@ -34,48 +36,47 @@ class Chip extends React.PureComponent<Props, any> {
       theme,
     } = this.props;
 
-    let className;
+    const className = classNames(
+      theme.Chip,
+      transparent && theme.transparent
+    );
 
-    if (this.props.markedForDelete) {
-      className = classNames(
-        theme.Chip,
-        theme.markedForDelete,
-        transparent && theme.transparent
-      );
-    } else {
-      className = classNames(
-          theme.Chip,
-          transparent && theme.transparent
-      );
-    }
+    const deleteInstruct = this.props.removable ? 'Press delete to remove this chip' : '';
 
     const chipContents = [(
       image
       ?
-      <img className={theme.Image} src={image.url} alt={image.alt} key="1" />
+      <img className={theme.Image} src={image.url} aria-hidden key="1" />
       : ''
     ),
       <span key="2">
         {this.props.children}
+      <span className={theme.hidden}>{deleteInstruct}</span>
       </span>,
     ];
-
     return (
-      <div className={className}>
+      <div
+        className={className}
+        tabIndex={this.props.tabIndex === 0 ? this.props.tabIndex : -1}
+        onFocus={this.props.autoSuggestMethods ? this.props.autoSuggestMethods.onfocus : ''}
+        onBlur={this.props.autoSuggestMethods ? this.props.autoSuggestMethods.onFocusOut : ''}
+        ref={this.props.autoSuggestMethods ? this.props.autoSuggestMethods.storeFocus : ''}
+        role="option"
+      >
         {
           clickable
             ?
-            <a onClick={this.props.onClick} aria-disabled={false} tabIndex={0}>
-                {chipContents}
+            <a onClick={this.props.onClick ? this.props.onClick : this.props.autoSuggestMethods.onClick} aria-disabled={false} role={'alert'} tabIndex={!this.props.tabIndex ? 0 : -1}>
```

Comment:
This onClick implementation logic is also not clean. I would suggest to have an onClick function in this component, which then calls the other logic.